File: api/libchams.py
import urllib.request

GITHUB_RAW = "https://github.com/FORTY71/neweposi/raw/refs/heads/main/libchams.so"


def handler(request):
    try:
        req = urllib.request.Request(GITHUB_RAW, headers={"User-Agent": "curl/8.0"})
        with urllib.request.urlopen(req, timeout=20) as r:
            data = r.read()
    except Exception:
        return {"statusCode": 502, "body": "upstream fetch failed"}
    return {
        "statusCode": 200,
        "headers": {
            "Content-Type": "application/octet-stream",
            "Content-Length": str(len(data)),
            "Cache-Control": "no-store",
        },
        "body": data,
        "isBase64Encoded": False,
    }

File: api/libchams.js
GITHUB_RAW = "https://github.com/FORTY71/neweposi/raw/refs/heads/main/libchams.so"

export default async function handler(request, response) {
  try {
    const r = await fetch(GITHUB_RAW, {
      headers: { "User-Agent": "Mozilla/5.0", Accept: "*/*" },
      redirect: "follow",
    })
    if (!r.ok) {
      response.status(502).send("upstream fetch failed: " + r.status)
      return
    }
    const buf = Buffer.from(await r.arrayBuffer())
    response.setHeader("Content-Type", "application/octet-stream")
    response.setHeader("Content-Length", buf.length)
    response.setHeader("Content-Disposition", 'attachment; filename="libchams.so"')
    response.setHeader("Cache-Control", "no-store")
    response.status(200).send(buf)
  } catch (e) {
    response.status(502).send("upstream fetch failed: " + e.message)
  }
}
